src/Workspaces/CoreTest/SolutionTests/SolutionTests.cs

```diff
@@ -2747,15 +2747,16 @@ public async Task TestAddingEditorConfigFileWithDiagnosticSeverity([Combinatoria
                     loader: TextLoader.From(TextAndVersion.Create(SourceText.From("[*.*]\r\n\r\ndotnet_diagnostic.CA1234.severity = error"), VersionStamp.Default)))));
 
             var newSyntaxTree = await solution.GetDocument(sourceDocumentId).GetSyntaxTreeAsync();
-            var newCompilation = await solution.GetProject(projectId).GetCompilationAsync();
+            var project = solution.GetProject(projectId);
+            var newCompilation = await project.GetCompilationAsync();
 
-            Assert.NotSame(originalSyntaxTree, newSyntaxTree);
+            Assert.Same(originalSyntaxTree, newSyntaxTree);
             Assert.NotSame(originalCompilation, newCompilation);
+            Assert.NotEqual(originalCompilation.Options, newCompilation.Options);
 
-            Assert.True(newCompilation.ContainsSyntaxTree(newSyntaxTree));
-
-            Assert.Single(newSyntaxTree.DiagnosticOptions);
-            Assert.Equal(ReportDiagnostic.Error, newSyntaxTree.DiagnosticOptions["CA1234"]);
+            var provider = project.CompilationOptions.SyntaxTreeOptionsProvider;
+            Assert.True(provider.TryGetDiagnosticValue(newSyntaxTree, "CA1234", out var severity));
+            Assert.Equal(ReportDiagnostic.Error, severity);
         }
 
         [Theory]
```

Comment:
I suspect this particular test is obsolete now, since it existed to make sure we were redoing the trees across both syntax tree types for C#/VB and for recoverable trees. It's not _bad_ but I'd probably just delete it. It's making sure we didn't screw up the old design, which is now moot.

Reply:
We're removed the useRecoverableTrees part, but we'll leave the rest.